<a href="https://colab.research.google.com/github/jumorap/discrete_mathematics_II_taller_III_alpha/blob/main/Taller3_disc_II.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

*Presented by: Juan C. Mora P. (jumorap@unal.edu.co)* \\
*Undergraduate in Computer Science (Systems Engineer)* \\

*Discrete Mathematics II UNAL* \\
*Francisco A. Gómez J.* \\

---

# Taller 3

De las notas de clase [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) y [notas de clase ecuaciones en diferencias segundo orden](https://drive.google.com/file/d/1CKgXegOOxv9Lx-3Uy8gMvTS32yixr78d/view?usp=sharing)

1. (10 Puntos) [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) - 14a - Punto 3
2. (10 Puntos) [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) - 14a - Punto 4
3. (10 Puntos) [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) - 14a - Punto 5
4. (10 Puntos) [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) - 14b - Punto 2
5. (10 Puntos) [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) - 14c - Punto 1 h
6. (10 Puntos) [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) - 14c - Punto 2 e
7. (10 Puntos) [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) - 14d - Punto 3
8. (10 Puntos) [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) - 14d  - Punto 7
9. (10 Puntos) [notas de clase ecuaciones en diferencias primer order](https://drive.google.com/file/d/1k67EShQw6JTmVU14iGVummO2kUwe8utH/view?usp=sharing) - 14e - Punto 4
10. (10 Puntos) [notas de clase ecuaciones en diferencias segundo orden](https://drive.google.com/file/d/1CKgXegOOxv9Lx-3Uy8gMvTS32yixr78d/view?usp=sharing) - 15b - Punto 4
11. (10 Puntos) [notas de clase ecuaciones en diferencias segundo orden](https://drive.google.com/file/d/1CKgXegOOxv9Lx-3Uy8gMvTS32yixr78d/view?usp=sharing) - 15.7 - Punto 7
12. (10 Puntos) [notas de clase ecuaciones en diferencias segundo orden](https://drive.google.com/file/d/1CKgXegOOxv9Lx-3Uy8gMvTS32yixr78d/view?usp=sharing) - 15.7 - Punto 8


# 1. 14a - 3) 
A vacuum pump removes one third of the remaining air in a cylinder with each stroke. Form an equation to represent this situation. After how many strokes is just $\frac{1}{1 000 000}$ of the initial air remaining?

Since each stroke removes $\frac{1}{3}$ of the air, then remaining $\frac{2}{3}$, and again is reduced by $\frac{1}{3}$, so, after n strokes we have left:<br>
$\Pi_{k = 0}^{n} \frac{2}{3} = \frac{2}{3} \cdot \frac{2}{3} \cdot \frac{2}{3} \cdot \frac{2}{3} \cdot \quad ... \quad \cdot \frac{2}{3}\quad$ $n$ times, until that $\Pi_{k = 0}^{n} \frac{2}{3} = \frac{1}{1000000}$

With this in mind, we can reduce the expression to an exponential expression, because the multiplying is not a dependent expression of $k$, as: $(\frac{2}{3})^{n}$, where $n$ tell us the number of times that shoud be stroked the vacuum to have $\frac{1}{1000000}$ of the total air. So:
<center>
    $(\frac{2}{3})^{n} = \frac{1}{1000000}$<br>
    $log_{\frac{2}{3}}((\frac{2}{3})^{n}) = log_{\frac{2}{3}}(\frac{1}{1000000})$<br>
    $n \cdot log_{\frac{2}{3}}(\frac{2}{3}) = log_{\frac{2}{3}}(\frac{1}{1000000})$<br>
    $n \cdot 1 = log_{\frac{2}{3}}(\frac{1}{1000000})$<br>
    $n = log_{\frac{2}{3}}(10^{-6})$<br>
    $n = -6 log_{\frac{2}{3}}(10) \approx 34$ times
</center>

# 2. 14a - 4)
Write down the first four terms of each of these sequences and the associated difference equation.<br>
(a) $u_{n} = \sum^{n}_{r=1} (2r - 1)$<br>
The first terms be: $1, 4, 9, 16$<br>
$n_{1} = 1$<br>
$n_{2} = 1 + (2*2 - 1)$<br>
$n_{3} = 1 + (2*2 - 1) + (2*3 - 1)$<br>
$n_{4} = 1 + (2*2 - 1) + (2*3 - 1) + (2*4 - 1)$<br>$\vdots$<br>
$n_{n-1} = 1 + (2*2 - 1) + (2*3 - 1) + (2*4 - 1) + \ ... \ + (2*n_{n-1} - 1)$<br>
$n_{n} = 1 + (2*2 - 1) + (2*3 - 1) + (2*4 - 1) + \ ... \ + (2*n_{n-1} - 1) + (2*n_{n} - 1)$<br>
$n_{n} = n_{n-1} + (2*n_{n} - 1)$

\\
(b) $u_{n} = \sum^{n}_{r=1} (10 - r)$<br>
The first terms be: $9, 17, 24, 30$<br>
$n_{1} = 9$<br>
$n_{2} = 9 + (10 - 2)$<br>
$n_{3} = 9 + (10 - 2) + (10 - 3)$<br>
$n_{4} = 9 + (10 - 2) + (10 - 3) + (10 -4)$<br>$\vdots$<br>
$n_{n-1} = 9 + (10 - 2) + (10 - 3) + (10 - 4) + \ ... \ + (10 - n_{n-1})$
$n_{n} = 9 + (10 - 2) + (10 - 3) + (10 - 4) + \ ... \ + (10 - n_{n-1}) + (10 - n_{n}$<br>
$n_{n} = n_{n - 1} + (10 - n_{n})$


\\
(c) $u_{n} = \sum^{n}_{r=1} 3(2r + 1)$<br>
The first terms be: $9, 15, 21, 27$<br>
$n_{1} = 9$<br>
$n_{2} = 9 + (6*2 + 3)$<br>
$n_{3} = 9 + (6*2 + 3) + (6*3 + 3)$<br>
$n_{4} = 9 + (6*2 + 3) + (6*3 + 3) + (6*4 + 3)$<br>$\vdots$<br>
$n_{n-1} = 1 + (6*2 + 3) + (6*3 + 3) + (6*4 + 3) + \ ... \ + (6*n_{n-1} + 3)$<br>
$n_{n} = 9 + (6*2 + 3) + (6*3 + 3) + (6*4 + 3) + \ ... \ + (6*n_{n-1} + 3) + (6*n_{n} + 3)$<br>
$n_{n} = n_{n-1} + (6*n_{n} + 3)$


# 3. 14a - 5) 
Write a simple computer program (say in Basic) which calculates successive terms of a sequence from a difference equation you have met. Here is one for the triangle numbers to help you.

*1O $\quad$ REM”SEQUENCE OF TRIANGLE NUMBERS”*<br>
*20 $\quad$ INPUT”NUMBER OF TERMS REQUIRED”;N*<br>
*30 $\quad$ CLS*<br>
*40 $\quad$ U=0:PRINT”FIRST “;N;” TRIANGLE NUMBERS ARE”*<br>
*50 $\quad$ FOR i=1 TO N:U=U+i:PRINTU:NEXTi*<br>
*60 $\quad$ STOP*

In [1]:
def lunch():
    try:
        n = int(input("Write the n number, where will the series end: "))
        print("The first n numbers be:")
        counter = 0

        for trig in range(n):
            counter += pow(trig + 1, 2)
            print(counter)
    except:
        print("Rectify your input")
        lunch()

lunch()

Write the n number, where will the series end: 100
The first n numbers be:
1
5
14
30
55
91
140
204
285
385
506
650
819
1015
1240
1496
1785
2109
2470
2870
3311
3795
4324
4900
5525
6201
6930
7714
8555
9455
10416
11440
12529
13685
14910
16206
17575
19019
20540
22140
23821
25585
27434
29370
31395
33511
35720
38024
40425
42925
45526
48230
51039
53955
56980
60116
63365
66729
70210
73810
77531
81375
85344
89440
93665
98021
102510
107134
111895
116795
121836
127020
132349
137825
143450
149226
155155
161239
167480
173880
180441
187165
194054
201110
208335
215731
223300
231044
238965
247065
255346
263810
272459
281295
290320
299536
308945
318549
328350
338350


# 4. 14b - 2)
A population is increasing at a rate of $25$ per thousand per year. Define a difference equation which describes this situation. Solve it and find the population in $20$ years' time, assuming the population is now $500$ million. How long will it take the population to reach $750$ million?

Taking the increasing as a percentage, we can define the equation as:
<center>
    $u_{n} = u_{n-1} \cdot 1.025$<br>
    $ u_{n-1} = u_{n-2} \cdot 1.025$<br>
</center>

So, we can define $u_{n}$ as:
<center>
    $u_{n} = (u_{n-2} \cdot 1.025) \cdot 1.025$<p>
    $u_{0} = 1000$<br>
    $u_{n} = u_{0} \cdot 1.025^{n}$
</center>

To $20$ years:
<center>
    $u_{0} = 500\ 000\ 000$<br>
    $u_{20} = u_{0} \cdot 1.025^{20}$<br>
    $u_{20} = 500\ 000\ 000 \cdot 1.025^{20} \approx 819\ 308\ 220 $
</center>

With this in mind, to a population of $750\ 000\ 000$ we have:
<center>
    $750\ 000\ 000 = 500\ 000\ 000 \cdot 1.025^{n}$<br>
    $(\frac{41}{40})^{n} = \frac{3}{2}$<p>
    $n = log_{\frac{41}{40}}(\frac{3}{2})$<p>
    $n \approx 16.4205 \rightarrow 16 - 17$ years
</center>


# 5. 14c - 1h) 
$u_{n} + 4u_{n - 1} + 3 = 0,$ $n \geq 1$ <br>
With the last expression, we can reduce as:
<center>
    $u_{n} + 4u_{n - 1} + 3 = 0$<br>
    $u_{n} = - 4u_{n - 1} - 3$
</center>

From this expression, is possible to obtain:
<center>
    $u_{n} = k^{n-1} + \frac{c(k^{n-1} - 1)}{k - 1}$<br>
    where $k = -4$,  $c = 3$<br>
    $u_{n} = -4^{n - 1} + \frac{-3(-4^{n - 1} - 1)}{-4 - 1}$<br>
    $u_{n} = -4^{n - 1} + \frac{-3(-4^{n - 1} - 1)}{-5}$<br>
    $u_{n} = -4^{n - 1}u_{1} + \frac{3}{5}(-4^{n - 1} - 1)$
</center>

# 6. 14c - 2e) 
$u_{0} = 3$ and $u_{n + 1} = u_{n} + 7$, $n \geq 0$<br>
Taking in mind the following expression:
<center>
    $u_{n} = u_{n - 1} + c$<br>
    $u_{n} = u_{0} + cn$
</center>

The statement will be worked in the following way:
<center>
    $u_{n + 1} = u_{n} + 7$<br>
    $u_{n} = u_{n - 1} + 7$<br>
    $u_{n} = u_{0} + 7n$, $\quad u_{0} = 3$<br>
    $u_{n} = 3 + 7n$
</center>

# 7. 14d - 3)
Find the monthly repayment on a $£400$ loan over a period of $1 \frac{1}{2}$ years at an interest rate of $15%$ p.a.

<center>
    $u_{n} = u_{n - 1} + (0.0125)u_{n - 1} - c$<br>
    $u_{n} = (1.0125)u_{n - 1} - c$
</center>

We have $k = 1.0125$, so:
<center>
    $u_{n} = 1.0125^{n} u_{0} - \frac{c(1.0125^{n} - 1)}{1.0125 - 1}$<br>
    $u_{n} = 1.0125^{n} u_{0} - 80c(1.0125^{n} - 1)$
</center>

When the period finish, $1$ and $\frac{1}{2}$ years, the value of $n = 18$
<center>
    $u_{18} = 1.0125^{n} u_{0} - 80c(1.0125^{18} - 1)$
</center>

Solving the variable $c$
<center>
    $c = (u_{18} - 1.0125^{18} \cdot 400) \cdot (-\frac{1.0125 - 1}{1.0125^{18} - 1})$
</center>

Having in mind that $u_{18}$ is $0$, because above the $u_{18}$ the loan was paid
<center>
    $c = (- 1.0125^{18} \cdot 400) \cdot (-\frac{1.0125 - 1}{1.0125^{18} - 1})$<p>
    $c \approx £24.95391 \approx  £24.95$
</center>


# 8. 14d - 7) 
A loan of $£1000$ is taken out at an interest rate of $24%$ p.a. How long would it take to repay the loan at a rate of $£50$ per month. [Note: this problem is similar to Question $3$, except that the value of $n$ cannot easily be found after you have solved the difference equation. A search technique on your calculator or a simple computer program will be necessary.]

For this exercise, we'll find n in the equation, as:
<center>
    $u_{n} = u_{n-1} + (0.02 \cdot u_{n-1}) - c$<br>
    $u_{n} = 1.02u_{n-1} - c$
</center>

And now, taking $u_{n}$ as $0$, because we wanna know how many time take to pay the loan (until reach 0), we use the expression:
<center>
    $0 = 1.02^{n} \cdot 1000 - \frac{50 \cdot (1.02^{n} - 1)}{1.02 - 1}$<br>
    $0 = 50((1.02^{n} \cdot 20) - 50(1.02^{n} - 1))$<br>
    $0 = (1.02^{n} \cdot 20) - 50(1.02^{n} - 1)$<br>
    $0 = (1.02^{n} \cdot 20) - 50 \cdot 1.02^{n} + 50$<br>
    $-50 = 1.02^{n}(-30)$<br>
    $\frac{5}{3} = 1.02^{n}$<p>
    $\frac{5}{3} = (\frac{51}{50})^{n}$<br>
    $n = log_{\frac{51}{50}}(\frac{5}{3})$<br>
    $x \approx 25.79585 \approx 26$ months
</center>

# 9. 14e - 4)
The productivity of an orchard of $2000$ trees increases by $5%$ each year due to improved farming techniques. The farmer also plants a further $100$ trees per year. Estimate the percentage improvement in productivity during the next $10$ years.

The behavior presented as an equation, taking in mind the sum of others $100$ trees, would be:
<center>
    $u_{n} = 1.05 \cdot u_{n - 1}  + 100$
</center>Where $u_{0} = 2000$, we have:<br>
$u_{1} = 1.05 \cdot 2000 + 100 = 2200$<br>
$u_{2} = 1.05 \cdot 2200 + 100 = 2410$<br>
$u_{3} = 1.05 \cdot 2410 + 100 = 2630.5$<br>
$u_{4} = 1.05 \cdot 2630.5 + 100 = 2862.025$

And now applying the general expression, we obtain:
<center>
    $u_{n} = (1.05^{n} \cdot u_{0} + \frac{100 \cdot (1.05^{n} - 1)}{1.05 - 1})$<br>
    $u_{n} = (1.05^{n} \cdot 2000 + \frac{100 \cdot (1.05^{n} - 1)}{1.05 - 1})$<br>
    To $u_{10}$ we have: $ \quad u_{10} = (1.05^{10} \cdot 2000 + \frac{100 \cdot (1.05^{10} - 1)}{1.05 - 1})$<br>
    $u_{10} = 4515.57851$
</center><p>So, to find the percentage:
<center><p>
    $u_{10_{percentage}} = \frac{u_{10}}{20} - 100$<br>
    $u_{10_{percentage}} = \frac{4515.57851}{20} - 100$<br>
    $u_{10_{percentage}} \approx 125.77892\% \approx 126\%$
</center>


# 10. 15B - 4)
Solve $ \ u_{n} - 6u_{n - 1} + 8u_{n - 2} = 0$, $n \geq 3$,  given $u_{1} = 10$, $u_{2} = 28$. Evaluate $u_{6}$

 First, we find the auxiliar equation, as:
 <center>
    $m^{2} - 6m + 8 = 0$<br>
    $(m - 4) \cdot (m - 2) = 0$<br>
    $m_{1} = 4 \quad \Lambda \quad m_{2} = 2$
 </center>

 So, right now, we can evaluate the equation as:
 <center>
    $u_{n} = A(m_{1})^{n} + B(m_{2})^{n}$<br>
    $u_{n} = A(4)^{n} + B(2)^{n}$<p>&nbsp;
    <p>
    $u_{1} = 10$ and $n = 1$<p>
    $10 = A \cdot 4 + B \cdot 2$<br>
    $5 = 2A + B$<br>
    $B = 5 - 2A$<p>&nbsp;
    <p>
    $u_{2} = 28$ and $n = 2$<p>
    $28 = A \cdot 16 + B \cdot 4$<br>
    $14 = 8A + 2B$<br>
    $A = 1$
 </center>

 Using the results in $B= 5 - 2A$, we obtain $B = 3$<p>
 So, the expression would be:
<center>
    $u_{n} = 4^{n} + 3 \cdot 2^{n}$<p>
    Now, $u_{6} = 4^{6} + 3 \cdot 2^{6} = 4288$
</center>

# 11. 15 7 - 7)
A difference equation of the form $u_{n} + au_{n - 1} + bu_{n - 2} = k$ defines a sequence with its first five terms as $0, 2, 5, 5, 14$. Find th $n$th term.

Using the formula $au_{n-1} + b u_{n-2} - k = -u_{n}$, we can generate the system:
<center><p>
$\left\{\begin{matrix}2a + 0b - k = -5\\ 
5a + 2b - k = -5\\
5a + 5b - k = -14
\end{matrix}\right.$
</center>
<p>And by Crammer, we find the $\Delta$

<center><p>
$\Delta_{0} = 
\begin{pmatrix}
    2 & 0 & -1\\
    5 & 2 & -1\\
    5 & 5 & -1
\end{pmatrix} = 9$<p>&nbsp;

$\Delta_{1} = 
\begin{pmatrix}
    -5 & 0 & -1\\
    -5 & 2 & -1\\
    -14 & 5 & -1
\end{pmatrix} = -18$<p>&nbsp;

$\Delta_{2} = 
\begin{pmatrix}
    2 & -5 & -1\\
    5 & -5 & -1\\
    5 & -14 & -1
\end{pmatrix} = -27$<p>&nbsp;

$\Delta_{3} = 
\begin{pmatrix}
    2 & 0 & -5\\
    5 & 2 & -5\\
    5 & 5 & -14
\end{pmatrix} = -81$<p>&nbsp;
</center>

So, the particular equations be:<br>
$a = \frac{18}{9} = 2$<br>
$b = -\frac{27}{9} = -3$<br>
$c = \frac{81}{9} = 9$<br>
And as result, we obtain the principal formula as:
<center>
    $u_{n} + 2 u_{n-1} + 3 u_{n-2} = 0$
</center>

Supposing that $u_{n-2} = a; u_{n-1} = a; u_{n} = a$
<center>
    $a + 2a - 3a = 9$<br> 
    $0 = 9$
</center>

With that result, we can deduce that the solution is inconsistent. And so, we can suppose that don't exist.

# 12. 15 7 - 8)
Find the smallest value of $n$ for which $u_{n}$ exceeds one million if $u_{n} = 10 + 3u_{n-1}$, $n \geq 1$, given $u_{0} = 0$

Applying the general expression, we obtain:

<center>
    $u_{n} = 3^{n} \cdot u_{0} + \frac{10 \cdot (3^{n} - 1)}{2}$<br>
    $u_{n} = 3^{n} \cdot u_{0} + 5 \cdot (3^{n} - 1)$<br>
    $u_{n} + 5 = 3^{n} \cdot (u_{0} + 5)$<br>
    $\frac{u_{n} + 5}{u_{0} + 5} = 3^{n}$
</center><p>So, we replace with $u_{n} = 1000000$ and $u_{0} = 0$
<center>
    $\frac{1000005}{5} = 3^{n}$<br>
    $n = log_{3}(200001)$<br>
    $n = \lceil 11.11045 \rceil = 12$<p>
    $u_{12} = 5 \cdot (3^{12} - 1) = 2657200$
</center>